servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/CloseHandler.java

```diff
@@ -260,6 +267,11 @@ public void protocolPayloadBeginOutbound(final ChannelHandlerContext ctx) {
 
         @Override
         public void protocolPayloadEndOutbound(final ChannelHandlerContext ctx) {
+            ctx.pipeline().fireUserEventTriggered(ProtocolPayloadEndEvent.OUTBOUND);
```

Comment:
I would avoid adding supposedly protocol specific logic like this to `UnsupportedProtocolHandler`. IIUC, this is to support H2 so lets create a `CloseHandler` implementation for H2 and for now we can simply implement this method.

Reply:
I assumed all protocols would need to provide some notion of payload completion for `WriteStreamSubscriber` to work properly, so it seemed reasonable to provide it here.